### PR TITLE
Enhancement/850/Redesining test cases to be clearer

### PR DESCRIPTION
**Pull request by AndreMarcel99**
##### SUMMARY
The zos_blockinfile and zos_lineinfile test cases were following a different design pattern over other test cases, this work item will align these test cases to other test cases and remove any duplicate or unnecessary tests. This also addresses a bug where one of the test cases was not cleaning up a data set it created.

Addresses issue #820

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
Include a new design for test case base on the test as it is en every test not separate, also the expected output.

##### ADDITIONAL INFORMATION
The set environment and remove environment functions are separate because are the repeated structure in almost every test case. The rest is included in test cases and for special cases Is added tu full creation of environment and remove in the test itself just in case of follow steps and do not lose that important information. Lastly try to remove all dictionaries were all was summary and add every parameters definition in the test. 

For next enhanced will be added test cases for the encoding because it was not correctly tested and added a extra combination for all test cases in data sets.

Captura de pantalla 2023-07-10 a la(s) 8 52 45

Captura de pantalla 2023-07-10 a la(s) 8 53 20

### CODE REVIEW

**Comment by AndreMarcel99** [comment]
Captura de pantalla 2023-07-12 a la(s) 15 42 20
Captura de pantalla 2023-07-12 a la(s) 15 43 30

Runs without any dictionaries

**Review by richp405 — APPROVED** [review]
The code change is significantly clearer, and the renaming of variables makes things better, as well.

**Review by ddimatos — CHANGES_REQUESTED** [review]
@AndreMarcel99 this is so much better than what we had before, I only made 2 comments but they apply to all test cases which are:
1) You can simplify the code if you want a bit by inspecting the stack to get the function name, its a bit expensive but probably worth it if we are reducing code and make it more flexible, eg `est_folder = "/tmp/ansible-core-tests/zos_blockinfile" + inspect.stack()[0][3]`
2) Move all the expected results into a variable at the top of the function is what the team decided was easier to read
3) Add 2 encoding tests , one for USS files and one for Data sets, i don't think we need more than 2 positive test cases, probably best to use the encoding module to help and then do the same for lineinfile.

**Review by rexemin — COMMENTED** [review]
Changes look really good, thank you for cleaning these tests up! I agree with Demetri's comments and don't really have anything else to add, so I'll approve when you make the changes.

File: tests/functional/modules/test_zos_blockinfile_func.py [hunk 1]
```diff
@@ -146,127 +106,72 @@
...
+    content = TEST_CONTENT
+    try:
+        set_uss_environment(ansible_zos_module, content, test_folder, test_name)
+        params["path"] = full_path
+        results = hosts.all.zos_blockinfile(**params)
+        for result in results.contacted.values():
+            assert result.get("changed") == 1
+        results = hosts.all.shell(cmd="cat {0}".format(params["path"]))
+        for result in results.contacted.values():
+            assert result.get("stdout") == """if [ -z STEPLIB ] && tty -s;
```
[the remark this answers is not in the document]
Comment (AndreMarcel99) [reply tests/functional/modules/test_zos_blockinfile_func.py]: The test for encoding of datasets, required specify texts

**Comment by AndreMarcel99** [comment]
Captura de pantalla 2023-07-17 a la(s) 9 16 39
New Run

File: tests/functional/modules/test_zos_blockinfile_func.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddimatos) [reply tests/functional/modules/test_zos_blockinfile_func.py]: Thanks for resolving the expected values into a variable.

**Comment by AndreMarcel99** [comment]
Captura de pantalla 2023-07-17 a la(s) 18 53 48

**Review by ddimatos — APPROVED** [review]
Looks good to me, thanks for all the hard work and the updates, this is an amazing cleanup.

**Pull request merged**